### ISSUE TITLE
Server crashes on pagination

### ISSUE THREAD

**Issue by coleedwards** [RyseInventory/RyseInventory#58]
Error: https://pastebin.com/YHGJNUbP
Code: https://pastebin.com/XM9RVHTw

Not sure how to fix, believe it's a bug.

**Comment by Rysefoxx** [comment]
Hey, 

it seems that it is not a simple error. The error just says that it calls my method Pagination#getPresent and from there on Java util methods are called.

I myself have tried many scenarios and could not reproduce the problem in my system.

1) Have you debugged your code and checked what values are floating around?
2) Have you already tried to comment out partial functions? Should it be because of a partial function, because it works without the method. Please debug this method further and give me possible values you find out.

I hope I can fix your problem in a timely manner.

**Comment by coleedwards** [comment]
I've had a day of debugging and change of code but can't seem to reproduce the error myself. It only occurs when a lot of players are online, using the GUIs. Could it be potentially too many items or something along that line?

**Comment by Rysefoxx** [comment]
Yes, I tried 10,000 items in one inventory. The server didn't crash itself, but it's most likely due to the fact that not many users have such a high number of items. However, the server does feel the strain.

This is most likely due to the fact that when you change the page all slots are recalculated.

I am working on a solution and will optimize this.
